packages/server/modules/webhooks/graph/resolvers/webhooks.js

```diff
@@ -0,0 +1,44 @@
+const { createWebhook, getWebhook, updateWebhook, deleteWebhook, getStreamWebhooks, getLastWebhookEvents, getWebhookEventsCount } = require( '../../services/webhooks' )
+
+
+module.exports = {
+  Stream: {
+    async webhooks( parent, args, context, info ) {
+      if ( args.id ) {
+        let wh = await getWebhook( { id: args.id } )
+        let items = wh ? [ wh ] : []
+        return { items, totalCount: items.length }
+      }
+
+      let items = await getStreamWebhooks( { streamId: parent.id } )
+      return { items, totalCount: items.length }
+    }
+  },
+
+  Webhook: {
+    async history( parent, args, context, info ) {
+      let items = await getLastWebhookEvents( { webhookId: parent.id, limit: args.limit } )
+      let totalCount = await getWebhookEventsCount( { webhookId: parent.id } )
+
+      return { items, totalCount }
+    }
+  },
+
+  Mutation: {
+    async webhookCreate( parent, args, context, info ) {
+      let id = await createWebhook( { streamId: args.webhook.streamId, url: args.webhook.url, description: args.webhook.description, secret: args.webhook.secret, enabled: args.webhook.enabled !== false, triggers: args.webhook.triggers } )
```

Comment:
We need to add the `await authorizeResolver( context.userId, args.stream.id, 'stream:owner' )` guard in here and everywhere else if it makes sense. 

We can discuss what's viewable by whom, but my £0.02 would be: 

- Stream contributors can see webhooks, but not create them
- Stream owners can see everything

If it's like this already, we're good (i didn't check!)

Reply:
re this, we previously decided that only `stream:owner`s could see / edit webhooks so that's how it is at the moment

if we want to change this, i can easily do it! note that right now, the settings page as a whole is not visible unless you are a `stream:owner`. i chose to do this bc editing stream details (the other settings page) is only available to owners, so the settings stuff would be entirely read only to a contributor

Reply:
conclusion: leave as is until poss more stuff is added to stream settings that would be relevant to contributors